### PR TITLE
feat: Automatically Create 'plugins.js/ts' in Config on New Strapi Project Initialization

### PR DESCRIPTION
**Pull request by L-Weisz**
### What does it do?

This update automatically creates a plugins.ts file in the ./config directory when initializing a new Strapi project. 
The file is generated with a basic template, providing a starting point for developers to configure their Strapi plugins.

### Why is it needed?

This update is born out of a personal pain point I encountered during Strapi development. I lost two days of development time and ended up creating an unnecessary issue in a Strapi plugin repository due to a simple mistake: I forgot to add an 's' to `plugins.ts` (wrote `plugin.ts`). This typo led to significant confusion and delay.

Despite the specific reference to `plugin.ts`, the mistake in the filename was overlooked when creating the issue and asking help to other developers , leading to prolonged troubleshooting and no resolution in sight. 
This experience highlighted a subtle yet critical point: such errors are easy to miss and can be difficult to spot even by others reviewing the problem.

By automatically generating the `plugins.ts` file during the initial project setup, this kind of error can be completely avoided.  Implementing this feature will kill this problem forever in a very simple yet effective way, greatly improving the developer experience and reducing the likelihood of similar issues in the future.

### How to test it?

Generate new strapi project 

### Related issue(s)/PR(s)

Didnt find any

### CODE REVIEW

**Comment by xaviermarchal** [comment]
@alexandrebodin Can you review this PR? seems like a good improvement for preventing from mistakes

**Comment by alexandrebodin** [comment]
Hey I think that would be fine. I'll check with our product team and add myself as a reviewer. (FYI we won't be making new release until eraly jan now :))

File: packages/generators/app/src/resources/files/js/config/plugins.js [hunk 1]
```diff
@@ -0,0 +1,2 @@
+module.exports = () => ({
```
Comment (derrickmehaffy) [inline packages/generators/app/src/resources/files/js/config/plugins.js:1]: The default should pass in the env function like some of the other config templates.

**Review by derrickmehaffy — CHANGES_REQUESTED** [review]

File: packages/generators/app/src/resources/files/js/config/plugins.js (hunk 1 above)
Comment (L-Weisz) [reply packages/generators/app/src/resources/files/js/config/plugins.js:1]: Hello @derrickmehaffy when I pass the env function without using it like that : 
```
export default ({ env }) => ({
});
```
 I then have unused variable error in CI, how should I solve this ?

Comment (derrickmehaffy) [reply packages/generators/app/src/resources/files/js/config/plugins.js:1]: I'm guessing that's something wrong with the linting this file should be excluded from that.

Any thoughts @alexandrebodin?

**Comment by L-Weisz** [comment]
Hello @derrickmehaffy I added the env variable but as you can see CodeFactor failed

File: packages/generators/app/src/resources/files/js/config/plugins.js (hunk 1 above)
Comment (L-Weisz) [reply packages/generators/app/src/resources/files/js/config/plugins.js]: Hello @alexandrebodin @derrickmehaffy any input on how to move forward with this ?

Comment (alexandrebodin) [reply packages/generators/app/src/resources/files/js/config/plugins.js]: the lint error is fine and the initial proposal is good for now. We should using typescript typings to help discovering that env param later but not necessary here

Comment (L-Weisz) [reply packages/generators/app/src/resources/files/js/config/plugins.js]: okay thanks I am modifying the PR to drop the last commits and go back to initial proposal

**Comment by L-Weisz** [comment]
@alexandrebodin as asked I went back to initial proposal and updated branch, thanks for you review

**Comment by alexandrebodin** [comment]
@L-Weisz can you target the develop branch please 🙏

**Comment by alexandrebodin** [comment]
Hey @L-Weisz  We have a little prettier issue on the PR can you run the `format:code` command and push again ?

**Comment by L-Weisz** [comment]
> Hey @L-Weisz We have a little prettier issue on the PR can you run the `format:code` command and push again ?

done , thanks again

**Review by alexandrebodin — APPROVED** [review]

**Pull request merged**